helios-core/src/main/kotlin/helios/syntax/json/syntax.kt

```diff
@@ -23,4 +25,22 @@ fun BigInteger.toJson(radix: Int): JsDecimal =
 
 fun BigDecimal.toJson(): JsDecimal = JsNumber.fromDecimalStringUnsafe(this.toString())
 
-//TODO add more syntax
\ No newline at end of file
+//TODO add syntax for all supported types
+
+fun <B> Json.asJsStringOrError(f: (JsString) -> Either<DecodingError, B>) =
```

Comment:
```suggestion
fun <B> Json.toJsStringOrError(f: (JsString) -> Either<DecodingError, B>) =
```

We're using `toJson` in other functions. We should be consistent with using `as` or `to`, and since `as` is also a keyword I would suggest `to`. This would also align it with Arrow.

Reply:
Yes, but `toJson` just parse it to, well, to a Json. On those methods, we are taking the `Json` and using it, `as` a `JsString` (or one of the other cases) and returning the function result (if the `Json` was actually a `JsString`)